code/object/objectdock.cpp

```diff
@@ -199,13 +200,13 @@ void dock_calc_docked_center_of_mass(vec3d *dest, object *objp)
 	dock_evaluate_all_docked_objects(objp, &dfi, dock_calc_docked_center_of_mass_helper);
 
 	// overall center of mass = weighted sum of centers of mass divided by total mass
-	vm_vec_scale(dest, (1.0f / dfi.maintained_variables.float_value));
+	float total_mass = dfi.maintained_variables.float_value;
+	vm_vec_scale(dest, (1.0f / total_mass));
+	return total_mass;
 }
 
 float dock_calc_total_docked_mass(object *objp)
 {
-	Assert(objp != NULL);
-
```

Comment:
Why was this removed?

Reply:
The function doesn't do anything that requires it to not be null in the body itself, but it calls `dock_evaluate_all_docked_objects` which does assert on `objp` being null. Not sure what the convention is here.

Reply:
I suppose that's reasonable.